main/segment.h

```diff
@@ -0,0 +1,89 @@
+#ifndef MAIN_SEGMENT_H
+#define MAIN_SEGMENT_H
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace micromouse::inline segment
+{
+
+inline constexpr auto epsilon = 1e-6f;
+
+using Line = Eigen::Hyperplane<float, 2>;
+
+// class Line
+// {
+// public:
+//     explicit constexpr Line(float a, float b, float c) noexcept
+//         : m_a{a}
+//         , m_b{b}
+//         , m_c{c}
+//         , m_slope{-a / b}
+//         , m_y_intercept{-c / b}
+//     {
+//     }
+//     explicit constexpr Line(float slope, float y_intercept) noexcept
+//         : m_a{-slope}
+//         , m_b{1}
+//         , m_c{-y_intercept}
+//         , m_slope{slope}
+//         , m_y_intercept{y_intercept}
+//     {
+//     }
+//     explicit constexpr Line(const Eigen::Vector2f &p1, const Eigen::Vector2f &p2)
+//         : m_a{p2(1) - p1(1)}
+//         , m_b{p1(0) - p2(0)}
+//         , m_c{-m_a * p1(0) - m_b * p1(1)}
+//         , m_slope{-m_a / m_b}
+//         , m_y_intercept{-m_c / m_b}
+//     {
+//     }
+
+//     constexpr auto a() const noexcept { return m_a; }
+//     constexpr auto b() const noexcept { return m_b; }
+//     constexpr auto c() const noexcept { return m_c; }
+//     constexpr auto slope() const noexcept { return m_slope; }
+//     constexpr auto y_intercept() const noexcept { return m_y_intercept; }
+
+// private:
+//     float m_a;
+//     float m_b;
+//     float m_c;
+
+//     float m_slope;
+//     float m_y_intercept;
+// };
+
+class Segment
+{
+public:
+    Segment(const Eigen::Vector2f &p1, const Eigen::Vector2f &p2) noexcept
+        : m_p1{p1}
+        , m_p2{p2}
+        , m_length{(p1 - p2).norm()}
+        , m_line(Line::Through(p1, p2))
+    {
+    }
+
+    auto p1() const noexcept { return m_p1; }
+    auto p2() const noexcept { return m_p2; }
+    constexpr auto length() const noexcept { return m_length; }
+    const auto &line() const noexcept { return m_line; }
```

Comment:
Maybe force lvalues here

```suggestion
    const auto &line() const & noexcept { return m_line; }
```